packages/gatsby/index.d.ts

```diff
@@ -620,7 +801,7 @@ export interface PluginOptions {
   [key: string]: unknown
 }
 
-export type PluginCallback = (err: Error | null, result?: any) => void
+export type PluginCallback<R = any> = (err: Error | null, result?: R) => void
```

Comment:
Do we need a generic here? Are we doing this because the type can be different, or because we just haven't done the research to discover the type here?